instance.tf

```diff
@@ -36,9 +36,19 @@ data "aws_ami" "amazon_linux_2" {
 resource "aws_instance" "fsb_instance" {
   ami                    = data.aws_ami.amazon_linux_2.id
   instance_type          = var.instance_type
-  key_name               = "MyKeyPair"
-  subnet_id              = aws_subnet.public[0].id
-  vpc_security_group_ids = [aws_security_group.FSB_SG.id]
+  key_name               = "OG"
+  subnet_id              = "subnet-0332aa50e8a8681cb"
```

Comment:
Could you please avoid hardcoding subnet_id id but rather use resource reference like you did before.